### PR TITLE
Adding model and partitioner for remote deployments

### PR DESCRIPTION
**Pull request by hhiden**
Created the model for splitting a flow over multiple deployments.

### CODE REVIEW

File: model/src/main/java/io/streamzi/openshift/dataflow/deployment/FlowDeploymentListBuilder.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import java.util.List;
+
+/**
+ * Creates a set of deployments using a ProcessorFlow
+ * @author hhiden
+ */
+public class FlowDeploymentListBuilder {
+    private ProcessorFlow flow;
+    private List<String> clouds = new ArrayList<>();
+    private String primaryCloudId = "local";
```
Comment (matzew) [inline model/src/main/java/io/streamzi/openshift/dataflow/deployment/FlowDeploymentListBuilder.java]: is that an `id` or a `name` ?

File: model/src/main/java/io/streamzi/openshift/dataflow/deployment/FlowDeployment.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            .forEach(node->nodeDeployments.add(new NodeDeployment(node, node.getTargetClouds().get(cloudId))));
+        
+
+        // Add the links that are not proxied
+        flow.getLinks().stream()
+            .filter(link->containsNode(link.getSource().getParent()))
+            .filter(link->containsNode(link.getTarget().getParent()))
+            .forEach(link->linkDeployments.add(new LinkDeployment(link, LinkDeployment.LinkType.INTERNAL_LINK, cloudId, cloudId)));
+                
+        // Add Proxies for the links that cannot be left as they are
```
Comment (matzew) [inline model/src/main/java/io/streamzi/openshift/dataflow/deployment/FlowDeployment.java]: not exactly sure I get the logic w/ the entire proxy here :-) 

Is that a local topic, that's "bridging" to something outside ?

Comment (hhiden) [reply model/src/main/java/io/streamzi/openshift/dataflow/deployment/FlowDeployment.java]: The idea is that links might be entirely local to a cloud or refer to a remote Kafka. Will Update link to go to a different bootstrap server

Comment (hhiden) [reply model/src/main/java/io/streamzi/openshift/dataflow/deployment/FlowDeployment.java]: All links will be the same - just have a bootstrap server and topic name

**Comment by matzew** [comment]
Can you rebase and/or cherry-pick?  This PR contains a commit that's already in the master branch, from Simon

**Pull request merged**